src/components/article-components/FullArticle.jsx

```diff
@@ -1,4 +1,46 @@
+import { useEffect, useState } from 'react';
+import { api } from '../../api';
+
 const FullArticle = ({ article }) => {
+	const [hasVoted, setHasVoted] = useState(false);
+	const [votes, setVotes] = useState(article.votes);
+
+	const handleUpvoteButton = e => {
+		api
+			.upvoteArticle(article.article_id, e.target.value)
+			.then(() => {
+				setVotes(current => {
+					return Number(current) + Number(e.target.value);
+				});
+
+				if (e.target.value == 1) {
+					setHasVoted(true);
+				} else {
+					setHasVoted(false);
+				}
+			})
+			.catch(err => {
+				alert('Could not upvote at this time, try again later.');
+			});
+	};
+
+	useEffect(() => {
+		const hasVotedFromStorage = localStorage.getItem(
+			`hasVoted-${article.article_id}`
+		);
+
+		if (hasVotedFromStorage) {
+			setHasVoted(JSON.parse(hasVotedFromStorage));
+		}
+	}, [article.article_id]);
+
+	useEffect(() => {
+		localStorage.setItem(
+			`hasVoted-${article.article_id}`,
+			JSON.stringify(hasVoted)
+		);
```

Comment:
this is an interesting way to approach this but using localStorage isn't the best way to go about this - i would think about how you could refactor this at a later date

Reply:
Absolutely agree